Sources/Bitcoin/PublicKey+Bitcoin.swift

```diff
@@ -9,8 +9,10 @@ import Foundation
 public extension PublicKey {
     /// Returns the public key address with the given prefix.
     public func bitcoinAddress(prefix: UInt8) -> BitcoinAddress {
-        let hash = Data([prefix]) + Crypto.sha256ripemd160(data)
-        return BitcoinAddress(data: hash)!
+        let witnessVersion = Data([0x00] + [0x14])
+        let redeemScript = Crypto.sha256ripemd160(witnessVersion + bitcoinKeyHash)
```

Comment:
Since this is specific to p2sh-p2wpkh is should be on a separate function: `bitcoinSegWitAddress`?